### PR TITLE
Sign-in with Ethereum

### PR DESCRIPTION
**Pull request by toastalot**
Implements EIP-4361 style sign-in with Ethereum
https://eips.ethereum.org/EIPS/eip-4361
https://login.xyz/

### CODE REVIEW

**Comment by AndrewYHuang** [comment]
I'm probably not the best person to review the web dev because I'm absolutely not a fan of Tailwind and Next.js, it does look like acceptable Tailwind and Next.js. You've successfully got a user from using ethers.js and metamask.

**Review by AndrewYHuang — DISMISSED** [review]
Mostly rants about frameworks, looks reasonable

File: tailwind.config.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+/** @type {import('tailwindcss').Config} */
+module.exports = {
+	content: [
+		'./src/pages/**/*.{js,ts,jsx,tsx}',
+		'./src/components/**/*.{js,ts,jsx,tsx}',
+		'./src/app/**/*.{js,ts,jsx,tsx}'
```
[the remark this answers is not in the document]
Comment (toastalot) [reply tailwind.config.js:6]: This was auto-generated by the Next CLI and for the purposes of this prototype I'm going to leave it in there for now

File: src/components/layout/Button.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+			onClick={action}
+			className={`text-white p-3 rounded-lg m-2 ${
+				dark ? 'bg-black border border-white' : 'bg-violet-700'
+			}`}
+			disabled={disabled}
+		>
+			{label}
+		</button>
+	)
+}
```
[the remark this answers is not in the document]
Comment (toastalot) [reply src/components/layout/Button.tsx:23]: This is a classic React pattern that allows you to pass props to change appearance based on some external logic e.g. if the global window.ethereum property exists then wallet is already connected so we disable the connect wallet button and make it dark, enable the login button and make it light

**Pull request merged**